modules/crop-ffmpeg/src/main/java/org/opencastproject/crop/impl/CropServiceImpl.java

```diff
@@ -0,0 +1,424 @@
+
+      return cropTrack;
+
+    } catch (Exception e) {
+      logger.warn("Error cropping video", e);
+      if (e instanceof CropException) {
+        throw (CropException) e;
+      } else {
+        throw new CropException(e);
+      }
+    }
+
+  }
+
+  protected Track cropFfmpeg(File mediafile, Track track) throws IOException, CropException {
+    String outputPath = FilenameUtils.removeExtension(mediafile.getAbsolutePath()).concat(RandomStringUtils
+            .randomAlphanumeric(8) + ".mp4");
+    String[] command = new String[] { binary, "-i", mediafile.getAbsolutePath(), "-vf",
+            "cropdetect=24:16:240", "-max_muxing_queue_size", "2000", "-y", outputPath };
+    String commandline = StringUtils.join(command, " ");
+
+    logger.info("Running {}", commandline);
+
+    ProcessBuilder pbuilder = new ProcessBuilder(command);
+    int cropValue = 0;
+    int widthVideo = 0;
+    String crop = null;
+    pbuilder.redirectErrorStream(true);
+    int exitCode = 1;
+    Process process = pbuilder.start();
+    BufferedReader errStream = new BufferedReader(new InputStreamReader(process.getInputStream()));
+    try {
+      String line = errStream.readLine();
+      while (null != line) {
+        if (line.startsWith("[Parsed_cropdetect")) {
+          // [Parsed_cropdetect_0 @ 0x8f6620] x1:120 x2:759 y1:0 y2:399 w:640 h:400 x:120 y:0 pts:639356 t:39.023193 crop=640:400:120:0
+          String[] lineSplitted = line.split(" ");
+          widthVideo = Integer.valueOf(lineSplitted[7].substring(2));
+          int x = Integer.valueOf(lineSplitted[9].substring(2));
+          if (cropValue == 0 || cropValue > x) {
+            cropValue = x;
+            crop = lineSplitted[13];
+          }
+
+        }
+        line = errStream.readLine();
+      }
+      exitCode = process.waitFor();
+
+    } catch (IOException e) {
+      logger.error("Error executing ffmeg: {}", e.getMessage());
+    } catch (InterruptedException e) {
+      logger.error("Waiting for encoder process exited was interrupted unexpected: {}", e.getMessage());
+    } finally {
+      IoSupport.closeQuietly(process);
+      IoSupport.closeQuietly(errStream);
+      if (exitCode != 0) {
+        try {
+          FileUtils.forceDelete(new File(outputPath));
+        } catch (IOException e) {
+
+        }
+      }
+    }
+
+    if (exitCode != 0) {
+      throw new CropException("The encoder process exited abnormally with exit code " + exitCode);
+    }
+    if (cropValue > widthVideo / 3) {
+      return track;
+    }
+    // FFMPEG-command for cropping video
+    logger.info("String for crop command: {}", crop);
+    String croppedOutputPath = FilenameUtils.removeExtension(mediafile.getAbsolutePath()).concat(RandomStringUtils
+            .randomAlphanumeric(8) + ".mp4");
+    String[] cropCommand = new String[] { binary, "-i", mediafile.getAbsolutePath(), "-vf", crop,
+            "-max_muxing_queue_size", "2000", "-y", croppedOutputPath };
```

Comment:
It would probably make sense to be able to configure the encoder options here. Possibly even using an encoding profile.